### ISSUE TITLE
Forgejo compatibility for `uses` lint

### ISSUE THREAD

**Issue by toonn** [rhysd/actionlint#698]
# Context

I'm using Actionlint for Forgejo Actions workflows.
[Forgejo's `uses` property allows specifying Actions with a full URL](https://forgejo.org/docs/v15.0/user/actions/actions/#using-actions).

# Example

The following is looked up relative to the `DEFAULT_ACTION_URL`, which is set to `https://data.forgejo.org` in my case.

```yaml
uses: dorny/paths-filter@v4
```

`https://data.forgejo.org/dorny/paths-filter@v4` doesn't exist resulting in an error. Forgejo mirrors some popular actions there, `actions/checkout` works transparently for example. As they can't mirror all of GitHub, they allow specifying a different base URL.

The following works for Forgejo's Actions.

```yaml
uses: https://github.com/dorny/paths-filter@v4
```

But now Actionlint complains:

```
.forgejo/workflows/nix-unit-module-schema.yaml:15:15: specifying action "https://github.com/dorny/paths-filter@v4" in invalid format because owner and repo and ref should not be empty. available formats are "{owner}/{repo}@{ref}" or "{owner}/{repo}/{path}@{ref}" [action]
   |
15 |         uses: https://github.com/dorny/paths-filter@v4
   |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

# Request

I'm not sure whether the project is interested in alternative Actions implementations but support for this broader format of the `uses` property would be great.

Potentially this could be gated on detecting whether an Action is a Forgejo Action somehow, the path containing `.forgejo` instead of `.github` for example.
